08-rails/18-testing-controllers.md

```diff
@@ -153,6 +154,26 @@ test "should create an entity" do
 end
 ```
 
+### What Kind of Thing Should You Test in the Controller?
+Controller tests are all about how your website responds to the input a user might give it. This includes a friendly user doing what they should, an idiot user banging into things, and a malicious user trying to break your site. This makes it a little different from the testing we've seen before.
+
+Exactly what's worth testing will come with time, but here are some general guidelines.
+- If your controller action reads a Model ID from the URL, you need at least 3 cases:
+  - The ID corresponds to a model in the DB
+  - The ID is not found in the DB
+  - The ID isn't a number
```

Comment:
If the ID isn't a number, it won't be found in the DB so it could be treated the same way.